bindings/pyroot/pythonizations/python/ROOT/_pythonization/_roofit/_roodatahist.py

```diff
@@ -50,3 +50,190 @@ def plotOn(self, *args, **kwargs):
         # The keywords must correspond to the CmdArg of the `plotOn` function.
         args, kwargs = _kwargs_to_roocmdargs(*args, **kwargs)
         return self._plotOn(*args, **kwargs)
+
+    @property
+    def shape(self):
+        import ROOT
+        import numpy as np
+
+        shape = []
+
+        var_is_category = self._var_is_category()
+
+        for i_var, var in enumerate(self.get()):
+            shape.append(var.numBins())
+
+        def get_len_from_shape():
+            n_size_check = 1
+            for n in shape:
+                n_size_check = n_size_check * n
+            return n_size_check
+
+        assert get_len_from_shape() == len(self)
+
+        return tuple(shape)
+
+    def __len__(self):
+        return self.numEntries()
+
+    def _to_array(self, buffer):
+        import numpy as np
+
+        # check if buffer is nullptr (for some reason comparing with ROOT.nullptr doesn't work)
+        try:
+            buffer[0]
+        except ReferenceError:
+            return None
+        a = np.copy(np.frombuffer(buffer, dtype=np.float64, count=len(self)))
+        return a.reshape(self.shape)
+
+    def _var_is_category(self):
+        """ """
```

Comment:
Do we want some docs here?

Reply:
Sure! Indeed it looked like I forgot that...